docs/changenodehostname.rst

```diff
@@ -0,0 +1,115 @@
+=======================================
+
+From the target node
+^^^^^^^^^^^^^^^^^^^^
+
+Change the hostname:
+
+.. code-block:: shell
+
+   [node]$ hostnamectl set-hostname <your hostname, here: ControlPlane2>
+   [node]$ systemctl restart systemd-hostnamed
+
+Check that the hostname has changed
+
+.. code-block:: shell
+
+   [node]$ hostnamectl status
+
+   Static hostname: controlplane2
+   Pretty hostname: ControlPlane2
+     Icon name: computer-vm
+      Chassis: vm
+   Machine ID: 5003025f93c1a84914ea5ae66519c100
+      Boot ID: f28d5c64f06c48a3a775e24c4f03d00c
+      Virtualization: kvm
+   Oerating System: CentOS Linux 7 (Core)
+      CPE OS Name: cpe:/o:centos:centos:7
+          Kernel: Linux 3.10.0-957.12.2.el7.x86_64
+      Architecture: x86-64
+
+
+From the bootstrap node
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Verify the hostname change allowed a change of status on the bootstrap,
+the node has now a status ``NotReady``.
+
+.. code-block:: shell
+
+   [bootstrap]$ kubectl get nodes
+   NAME               STATUS     ROLES                         AGE       VERSION
+   node2.novalocal    NotReady   etcd,master                   19h       v1.11.7
+
+Change the name of the node in its manifest file:
+
+.. code-block:: diff
+
+   apiVersion: v1
+   kind: Node
+   metadata:
+   -  name: old_hostname
+   +  name: new_hostname
+     annotations:
+       metalk8s.scality.com/ssh-key-path: /etc/metalk8s/pki/id_rsa
+       metalk8s.scality.com/ssh-host: 172.21.254.46
+       metalk8s.scality.com/ssh-sudo: 'false'
+     labels:
+       metalk8s.scality.com/version: '2.0'
+       node-role.kubernetes.io/master: ''
+       node-role.kubernetes.io/etcd: ''
+   spec:
+     taints:
+     - effect: NoSchedule
+       key: node-role.kubernetes.io/master
+     - effect: NoSchedule
+       key: node-role.kubernetes.io/etcd
+
+
+Then apply the configuration:
+
+.. code-block:: shell
+
+   [bootstrap]$ kubectl apply -f new-node-master3.yaml
+
+Delete the previous definition of the node (here ``node2.novalocal``)
+in Kubernetes:
+
+.. code-block:: shell
+
+   [bootstrap]$ kubectl delete node node2.novalocal
+
+Obtain a shell into the Salt master container:
+
+.. code-block:: shell
+
+   [bootstrap]$ kubectl -it -n kube-system exec salt-master-master.novalocal bash
+
+Execute the node deployment procedure with the node's new name:
+
+.. code-block:: shell
+
+   [salt-master]$ salt-run state.orchestrate metalk8s.orchestrate.deploy_node saltenv=metalk8s-2.0 pillar="{'orchestrate': {'node_name': 'controlplane2'}}"
+
+
+   Summary for master.novalocal_master
+   -------------
+   Succeeded: 11 (changed=9)
+   Failed:     0
+   -------------
+   Total states run:     11
+   Total run time:  132.435 s
+
+
+On the updated node, delete the now invalid certificates:
+
+.. code-block:: shell
+
+   [bootstrap]$ rm -f /var/lib/kubelet/pki/*.pem
+
+Then restart the kubelet service:
+
+.. code-block:: shell
+
+   [bootstrap]$ systemctl restart kubelet
```

Comment:
(not related to this line)

@NicolasT also mentioned (e-mail) that we should remove the previous minion key from the master (before deployment as well would be good)